prow/jobs/custom/infra.yaml

```diff
@@ -0,0 +1,197 @@
+# Copyright 2023 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Add a fake hook image here so that autobump.sh can bump only job configs
+# image: gcr.io/k8s-prow/hook:v20230329-c93d79fb7d
+# See
+# https://github.com/kubernetes/test-infra/blob/5815354584709c3f436e3d682110c673d224d7b1/prow/cmd/autobump/autobump.sh#L164
+
+presubmits:
+  knative/infra:
+  - name: pull-infra-validate-prow-yaml
+    run_if_changed: '^prow/((config|plugins)\.yaml$|jobs/)'
+    decorate: true
+    cluster: prow-build
+    spec:
+      containers:
+      - image: gcr.io/k8s-prow/checkconfig:v20230329-c93d79fb7d
+        command:
+        - checkconfig
+        args:
+        - --config-path=./prow/config.yaml
+        - --job-config-path=./prow/jobs
+        - --plugin-config=./prow/plugins.yaml
+        - --strict
+        # This warning can forbid valid (and convenient) config. Exclude it.
+        # Details: https://github.com/GoogleCloudPlatform/oss-test-infra/pull/131#discussion_r334208082
+        - --exclude-warning=mismatched-tide
+        - --exclude-warning=non-decorated-jobs
+        - --exclude-warning=long-job-names
+
+  - name: pull-infra-validate-own-testgrid-yaml
+    decorate: true
+    optional: false
+    run_if_changed: "(^config/prow/k8s-testgrid/.*.yaml$|^prow/jobs/.*.yaml$)"
+    branches:
+    - "main"
+    cluster: "prow-build"
+    decoration_config: # will remove this temp override
+      gcs_credentials_secret: "gcs-upload"
+    extra_refs:
+    - org: GoogleCloudPlatform
+      repo: oss-test-infra
+      base_ref: master
+    spec:
+      containers:
+      - image: gcr.io/k8s-prow/configurator:v20230329-c93d79fb7d
+        command:
+        - configurator
+        args:
+        - --prow-config=prow/config.yaml
+        - --prow-job-config=prow/jobs
+        - --default=config/prow/k8s-testgrid/testgrid-common-settings.yaml
+        - --yaml=config/prow/k8s-testgrid/k8s-testgrid.yaml
+        - --oneshot
+        - --validate-config-file
+
+  - name: pull-knative-infra-prow-tests-image-build
+    agent: kubernetes
+    optional: false
+    decorate: true
+    path_alias: knative.dev/infra
+    cluster: prow-build
+    run_if_changed: "^(images/prow-tests/.*|tools/.*|pkg/.*|go.mod)$"
```

Comment:
I think there's other question. Previously, those jobs were run when the code of those tools have changed. We have a couple of options here:

1. Reference `knative.dev/toolbox` in tools.go, so it will be tracked in go.mod (my fav)
2. Install specific commit of a tool with `go install ...@deadbeef`
3. Run those jobs manually. Or trigger them from toolbox, somehow.

Reply:
we can trigger from infra (as we have not configured .repos yet), but this should be under toolsbox's own job not part of infras.

Reply:
I will make the comment on #53 then